fendermint/testing/materializer/src/manifest.rs

```diff
@@ -0,0 +1,223 @@
+        validators: CollateralMap,
+        /// Balances of the accounts in the rootnet.
+        ///
+        /// These balances will go in the genesis file.
+        balances: BalanceMap,
+        /// Nodes that participate in running the root chain.
+        nodes: NodeMap,
+    },
+}
+
+/// An IPC subnet.
+///
+/// The balance of the account on the parent subnet, as declared in this manifest,
+/// _does not_ have to account for the collateral/balance we have to take from it to join/fund the subnet.
+/// When we create the testnet configuration from the manifest we will account for this with a rollup,
+/// so we don't have to do that much mental arithmetic and run into frustrating errors during setup.
+/// If we want to test trying to join with more than what we have, we can do so in the integration test.
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+pub struct Subnet {
+    /// The account we use to create the subnet.
+    pub creator: AccountId,
+    /// Collateral of the initial validator set.
+    ///
+    /// These validators will join the subnet with these collaterals after the subnet is created.
+    pub validators: CollateralMap,
+    /// Balances of the accounts at the creation of the subnet.
+    ///
+    /// These accounts will pre-fund the subnet after it's created.
+    pub balances: BalanceMap,
+    /// Nodes that participate in running the chain of this subnet.
+    pub nodes: NodeMap,
+    /// Relayers that submit bottom-up checkpoints to the parent subnet.
+    pub relayers: RelayerMap,
+    /// Child subnets under this parent.
+    ///
+    /// The subnet ID exists so we can find the outcome of existing deployments in the log.
+    pub subnets: SubnetMap,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+pub struct Node {
+    /// Indicate whether this is a validator node or a full node.
+    pub mode: NodeMode,
+    /// Indicate whether to run the Ethereum API.
+    pub ethapi: bool,
+    /// The nodes from which CometBFT should bootstrap itself.
+    pub seed_nodes: Vec<NodeId>,
+    /// The parent node that the top-down syncer follows;
+    /// or leave it empty if the parent is CalibrationNet.
+    pub parent_node: Option<NodeId>,
+}
+
+/// The mode in which CometBFT is running.
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+pub enum NodeMode {
+    /// A node able to create and sign blocks.
+    Validator(AccountId),
+    /// A node which runs consensus and executes blocks, but doesn't have a validator key.
+    Full,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+pub struct Relayer {
+    /// The account which will pay for the submission on the parent subnet.
+    pub submitter: AccountId,
+    /// The node which the relayer is following on the subnet.
+    pub follow_node: NodeId,
+    /// The node where the relayer submits the checkpoints;
+    /// or leave it empty if the parent is CalibrationNet.
+    pub submit_node: Option<NodeId>,
```

Comment:
How would we handle the case of an external parent? e.g. Calibnet?

Reply:
Currently I thought the tester can just know the load balanced glif.io endpoint, but we can make it an explicit part of the manifest using a JSON-RPC endpoint, configured individually on any node that talks to an external address.

Reply:
Added a `ParentNode` type for this.